template/authz/template.proto

```diff
@@ -0,0 +1,27 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package authz;
+
+import "mixer/v1/config/descriptor/value_type.proto";
+import "mixer/v1/template/extensions.proto";
+
+option (istio.mixer.v1.template.template_variety) = TEMPLATE_VARIETY_CHECK;
+
+message Template {
+  map<string, istio.mixer.v1.config.descriptor.ValueType> subject = 1;
+  map<string, istio.mixer.v1.config.descriptor.ValueType> action = 2;
```

Comment:
Does action really need to be this generic? Wouldn't a single string be sufficient here?

Reply:
According to the Istio Native RBAC Design V2 document, action has multiple fields, namespace, service, method, path and attributes.